### ISSUE TITLE
TEI Header - Špecifikácia

### ISSUE THREAD

**Issue by psekan** [LIBCAS/DL4DH#25]
V nadväznosti na issue #19 je potrebné definovať, aký obsah má mať element `teiHeader` a odkiaľ majú byť čerpané dáta do jeho podčastí. Momentálne má element nasledujúcu štruktúru:

```xml
  <teiHeader>
    <fileDesc>
      <titleStmt>
        <title>${title}</title>
        <author>
          <orgName>${orgName}</orgName>
          <persName>${persName}</persName>
          <idno type="mods:nameIdentifier">${nameIdentifier}</idno>
        </author>
      </titleStmt>
      <extent>${extent}</extent>
      <publicationStmt>
        <publisher>${publisher}</publisher>
        <pubPlace>${place}</pubPlace>
        <date>${date}</date>
{for identifiers as type => value}
        <idno type="mods:${type}">${value}</idno>
{/for}
        <availability>
          <licence resp="#NameTag">CC BY-NC-SA</licence>
        </availability>
        <availability>
          <licence resp="#UDPipe">CC BY-NC-SA</licence>
        </availability>
      </publicationStmt>
      <sourceDesc>
        <bibl/>
      </sourceDesc>
    </fileDesc>
    <profileDesc>
      <textClass>
        <classCode scheme="https://ufal.mff.cuni.cz/nametag/2/models">
          <interpGrp>
...
          </interpGrp>
        </classCode>
      </textClass>
      <langUsage>
        <language ident="${lang}">${lang}</language>
      </langUsage>
    </profileDesc>
  </teiHeader>
  ```
  
  Obsah elementu `interpGrp` je fixný podľa ukážky TEI od @daliboris v #19 .
  Namiesto výrazov `${...}` sa vkladajú príslušne hodnoty, ktoré sú momentálne vyťahované z formátu MODS. Tie sme dedukovali pomocou reverzného inžinierstva zo spomínanej ukážky nasledovne:
  
  | Výraz | element z MODS formátu |
  |---|---|
  | `title` | `/mods/titleInfo/title` |
  | `orgName` | `/mods/name/namePart` - vypĺňa sa len ak `/mods/name[@type='corporate']`|
  | `persName` | `/mods/name/namePart` - vypĺňa sa len ak `/mods/name[not(@type='corporate')]`|
  | `nameIdentifier` | `/mods/name/nameIdentifier` |
  | `extent` | `/mods/physicalDescription/extent` |
  | `publisher` | `/mods/originInfo/publisher` |
  | `place` | `/mods/originInfo/place/placeTerm[@type='text']` |
  | `date` | `/mods/originInfo/dateIssued` |
  | `identifiers` | vytiahnuté pole údajov z elementov `/mods/identifier` |
  | `lang` | `/mods/language/languageTerm` |
  
  Prosíme o potvrdenie, či:
  - je aktuálna verzia správna a konečná, alebo sa bude ešte rozširovať o nejaké elementy
  - údaje sú správne vyťahované z MODS formátu a zo správnych častí tohto formátu, alebo má byť využitý iný formát (FOXML, MARC, ...)
  - ak z MODS danú hodnotu nie sme schopní vytiahnuť (napríklad element `/mods/physicalDescription/extent` neexistuje), má byť element v TEI, do ktorého sa daná hodnota dopĺňa prázdny, alebo odmazaný
  - atribút `ident` elementu `language` má mať rovnakú hodnotu ako obsah daného elementu
  - element `fileDesc/sourceDesc/bibl` má byť prázdny?

**Comment by bodnarIQ** [comment]
Doplňujúci dotaz ohľadom elementu `/mods/identifier`

V MODS pre publikáciu s ID: `uuid:319546a0-5a42-11eb-b4d1-005056827e51` - _Protokol ... veřejné schůze bratrstva sv. Michala v Praze dne_ majú niektoré elementy identifikátorov atribút `invalid="yes"`, inde naopak chýbajú. Podľa xsd špecifikácie MODS je ale default pre atribút `invalid` hodnota "yes", takže pokiaľ tento atribút chýba, nastaví sa na "yes". 

<details>
<summary>Ukážka</summary>

```xml
        <mods:identifier type="barCode">1004407716</mods:identifier>
        <mods:identifier type="uuid">319546a0-5a42-11eb-b4d1-005056827e51</mods:identifier>
        <mods:identifier invalid="yes" type="ccnb">cnb000964752</mods:identifier>
        <mods:identifier invalid="yes" type="ccnb">cnb001503907</mods:identifier>
        <mods:identifier type="oclc">85591009</mods:identifier>
        <mods:identifier type="oclc">85001648</mods:identifier>
```
</details>

Súdiac z toho, že atribút je definovaný iba s hodnotou "yes" sa domnievam, že to bolo myslené tak, že ak tam ten atribút nie je, tak bude default "no". Do Krameria+ sa ale tým pádom bude dostávať stále hodnota "yes". Neviem, či je to atribút, na ktorom nejak záleží, prípadne či sa bude nejak ďalej využívať, ale vnímam to ako možnú chybu pri vytváraní MODS metadát.

**Comment by daliboris** [comment]
Na stránce s [dokumentací](https://www.loc.gov/standards/mods/userguide/identifier.html) je tato poznámka:

>  If invalid="yes" is not present, the identifier is assumed to be valid. Do not use invalid="no."

Máte pravdu, že tomu neodpovídá definice v [XSD](https://www.loc.gov/standards/mods/mods.xsd), ale vycházejme raději z výše uvedené poznámky.

Myslím, že nevalidní identifikátory bychom do výstupu v TEI nemuseli zahrnovat.

**Comment by bodnarIQ** [comment]
V tom prípade by som navrhoval riešiť to úpravou defínice XSD pre naše potreby. XSD používam pre automatické generovanie tried a v jej momentálnej podobe vracia atribút `invalid` hodnotu "yes" v oboch prípadoch (aj ak je atribút definovaný s hodnotou "yes" , a aj ak atribút nie je v elemente definovaný)

**Comment by daliboris** [comment]
Souhlasím, z hlediska XSD bude mít nejblíže našemu záměru tato definice atributu:

```xml
   <xs:attribute name="invalid" use="optional">
    <xs:simpleType>
     <xs:restriction base="xs:NMTOKEN">
      <xs:enumeration value="yes" />
     </xs:restriction>
    </xs:simpleType>
   </xs:attribute>
```

Doporučuju ale, abyste tuto úpravu původní XSD nějak automatizovali, např. pomocí XSLT, aby se na to při aktualizaci vašeho softwaru při přechodu na novější verzi MODS nezapomnělo.

**Comment by daliboris** [comment]
Do elementu `<publicationStmt>` je potřeba ještě doplnit element `<availability>`, který se bude vztahovat na celý text publikace.

Ukázka:
 ```xml
<availability status="free">
  <licence>[????není jasné, pod jakou licencí jsou volná díla zpřístupněna???]</licence>
</availability>
```

Hodnota atributu `@status` bude záležet na hodnotě elementu `<dc:rights>`

 ```xml
<oai_dc:dc xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:oai_dc="http://www.openarchives.org/OAI/2.0/oai_dc/">
<dc:rights>policy:public</dc:rights>
</oai_dc:dc>
```

TEI [definuje](https://tei-c.org/release/doc/tei-p5-doc/en/html/ref-availability.html) pro tento atribut hodnoty **free**, **unknown** a **restricted**.

Upřesňující informace by měla být také součástí elementu `<licence>`, ale to bude záležet nejspíš na jednotlivých knihovnách, tj. jakým způsobem licenci pro konkrétní text nastaví (pro ne/komerční použití ap.). Nevím, odkud tuto informaci přebírat.
